torchtune/modules/multimodal_transformer.py

```diff
@@ -0,0 +1,206 @@
+# Copyright (c) Meta Platforms, Inc. and affiliates.
+# All rights reserved.
+#
+# This source code is licensed under the BSD-style license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch
+from torch import nn, Tensor
+
+from torchtune.modules import GroupedQueryAttention
+
+
+class TransformerSelfAttentionLayer(nn.Module):
```

Comment:
Would be good to just rename this tp have "Multimodal" in the prefix

Reply:
This is also used in clip. I think that we should only use MM for modules that touch at the same time text and image. This is not the case for this module. What do you think?